src/cursor/abstract_cursor.ts

```diff
@@ -604,7 +624,11 @@ function next(cursor: AbstractCursor, callback: Callback<Document | null>): void
       return cleanupCursor(cursor, () => callback(err, nextDocument(cursor)));
     }
 
-    next(cursor, callback);
+    if (cursor[kDocuments].length === 0 && blocking === false) {
+      return callback(undefined, null);
```

Comment:
Is there a specific reason to use `null` over `undefined`? I suppose it makes more sense to communicate a non-existent object, so I'm not taking issue with it just wondering if there was some thinking behind it

Although maybe one reason to use undefined is so the result doesn't respond true to: `typeof result === 'object'`.

Reply:
I used `null` here for symmetry with our semantics around a "dead" cursor. When a cursor returns a zero cursor id, we return an explicit `null` to the user. I think this might have been left over from days when a `Cursor` inherited `Readable` since ending a stream is accomplished by `readable.push(null)`, but either way the convention exists so I stuck with it